File: docker/Dockerfile.java
# Deprecated: use apps/java-enterprise/Dockerfile from repo root context.
FROM maven:3.9.9-eclipse-temurin-21 AS builder

WORKDIR /src
COPY apps/java-enterprise/pom.xml ./
RUN mvn -q -DskipTests dependency:go-offline

COPY apps/java-enterprise/src ./src
RUN mvn -q -DskipTests package

FROM eclipse-temurin:21-jre-alpine
WORKDIR /app
RUN addgroup -S spring && adduser -S spring -G spring
COPY --from=builder /src/target/java-enterprise-0.1.0.jar /app/app.jar
USER spring:spring
EXPOSE 4004
ENTRYPOINT ["java", "-jar", "/app/app.jar"]

File: docker/Dockerfile.go
FROM golang:1.24-alpine AS builder

WORKDIR /src
# Pre-download dependencies to leverage Docker cache
COPY apps/go-orchestrator/go.mod apps/go-orchestrator/go.sum ./
RUN go mod download

# Build the binary
COPY apps/go-orchestrator ./
RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /out/go-orchestrator ./cmd

# Final runtime image
FROM alpine:3.20
RUN apk add --no-cache ca-certificates
RUN adduser -D -H -s /sbin/nologin appuser
USER appuser

WORKDIR /app
COPY --from=builder /out/go-orchestrator /app/go-orchestrator

EXPOSE 4001 50051
ENTRYPOINT ["/app/go-orchestrator"]
